File: Binary search tree (BST)/Kth Smallest Element in a BST/Kth Smallest Element in a BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> v;
    void storeTree(TreeNode* root){
        if( root == NULL) return;
        storeTree(root->left);
        v.push_back(root->val);
        storeTree(root->right);
    }
    int kthSmallest(TreeNode* root, int k) {
        if( root != NULL ) 
            storeTree(root);
        return v[k-1];
    }
};

File: Binary search tree (BST)/Kth Smallest Element in a BST/Kth Smallest Element in a BST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    ArrayList<Integer> v = new ArrayList<Integer>();
    void storeTree (TreeNode root ){
        if(root == null) return;
        storeTree(root.left);
        v.add(root.val);
        storeTree(root.right);
    }
    public int kthSmallest(TreeNode root, int k) {
          if( root != null ) 
            storeTree(root);
        return v.get( k - 1 );
    }
}
